full-stack-practice/client/src/Catalog.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react';
+import { type Product } from './data';
+import { Link } from 'react-router-dom';
+import { toDollars } from './to-dollars';
+
+export function Catalog() {
+  const [products, setProducts] = useState<Product[]>();
```

Comment:
You assume that if `products` is not defined then it is loading. That isn't true if there is a network error. In that case, you will have already determined there's a problem, but `products` will be undefined, so you'll display a Loading message. That can be confusing to the user.

This is a nice thought, but I think the `isLoading` state is the best way to build this.